lib/charms/magma_orchestrator_interface/v0/magma_orchestrator_interface.py

```diff
@@ -0,0 +1,237 @@
+        print(event.fluentd_port)
+
+
+if __name__ == "__main__":
+    main(DummyMagmaOrchestratorRequirerCharm)
+```
+"""
+
+
+import logging
+
+from jsonschema import exceptions, validate  # type: ignore[import]
+from ops.charm import CharmBase, CharmEvents, RelationChangedEvent
+from ops.framework import EventBase, EventSource, Handle, Object
+
+# The unique Charmhub library identifier, never change it
+LIBID = "ec30058c7c6d4850aba6a132d2506efe"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+
+logger = logging.getLogger(__name__)
+
+REQUIRER_JSON_SCHEMA = {
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "`magma-orchestrator` requirer root schema",
+    "type": "object",
+    "description": "The `magma-orchestrator` root schema comprises the entire requirer databag for this interface.",  # noqa: E501
+    "examples": [
+        {
+            "root_ca_certificate": "-----BEGIN CERTIFICATE-----\nMIICvDCCAaQCFFPAOD7utDTsgFrm0vS4We18OcnKMA0GCSqGSIb3DQEBCwUAMCAx\nCzAJBgNVBAYTAlVTMREwDwYDVQQDDAh3aGF0ZXZlcjAeFw0yMjA3MjkyMTE5Mzha\nFw0yMzA3MjkyMTE5MzhaMBUxEzARBgNVBAMMCmJhbmFuYS5jb20wggEiMA0GCSqG\nSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDVpcfcBOnFuyZG+A2WQzmaBI5NXgwTCfvE\neKciqRQXhzJdUkEg7eqwFrK3y9yjhoiB6q0WNAeR+nOdS/Cw7layRtGz5skOq7Aa\nN4FZHg0or30i7Rrx7afJcGJyLpxfK/OfLmJm5QEdLXV0DZp0L5vuhhEb1EUOrMaY\nGe4iwqTyg6D7fuBili9dBVn9IvNhYMVgtiqkWVLTW4ChE0LgES4oO3rQZgp4dtM5\nsp6KwHGO766UzwGnkKRizaqmLylfVusllWNPFfp6gEaxa45N70oqGUrvGSVHWeHf\nfvkhpWx+wOnu+2A5F/Yv3UNz2v4g7Vjt7V0tjL4KMV9YklpRjTh3AgMBAAEwDQYJ\nKoZIhvcNAQELBQADggEBAChjRzuba8zjQ7NYBVas89Oy7u++MlS8xWxh++yiUsV6\nWMk3ZemsPtXc1YmXorIQohtxLxzUPm2JhyzFzU/sOLmJQ1E/l+gtZHyRCwsb20fX\nmphuJsMVd7qv/GwEk9PBsk2uDqg4/Wix0Rx5lf95juJP7CPXQJl5FQauf3+LSz0y\nwF/j+4GqvrwsWr9hKOLmPdkyKkR6bHKtzzsxL9PM8GnElk2OpaPMMnzbL/vt2IAt\nxK01ZzPxCQCzVwHo5IJO5NR/fIyFbEPhxzG17QsRDOBR9fl9cOIvDeSO04vyZ+nz\n+kA2c3fNrZFAtpIlOOmFh8Q12rVL4sAjI5mVWnNEgvI=\n-----END CERTIFICATE-----\n",  # noqa: E501
+            "orchestrator_address": "http://orchestrator.com",
+            "orchestrator_port": 1234,
+            "bootstrapper_address": "http://bootstrapper.com",
+            "bootstrapper_port": 5678,
+            "fluentd_address": "http://fluentd.com",
+            "fluentd_port": 9112,
+        }
+    ],
+    "properties": {
+        "root_ca_certificate": {
+            "type": "string",
+        },
+        "orchestrator_address": {
+            "type": "string",
+        },
+        "orchestrator_port": {
+            "type": "integer",
+        },
+        "bootstrapper_address": {
+            "type": "string",
+        },
+        "bootstrapper_port": {
+            "type": "integer",
+        },
+        "fluentd_address": {
+            "type": "string",
+        },
+        "fluentd_port": {
+            "type": "integer",
```

Comment:
Moved to using strings because relation data values can only be strings. So those won't be usable unfortunately.

Reply:
How about keeping integers here, with additional constraints, for validation purposes and casting to string while pushing to the data bag?

Reply:
The problem with this is that the requirer would have to manipulate the relation data before validating it with json schema. Not sure if that's better. We can enforce those values at the interface level for the provider instead of in the relation data.